sql/core_clients_last_seen_raw_v1.sql

```diff
@@ -0,0 +1,62 @@
+  -- Equivalent to, but more efficient than, calling udf_bitmask_range(1, 28)
+CREATE TEMP FUNCTION
+  udf_bitmask_lowest_28() AS (0x0FFFFFFF);
+  --
+CREATE TEMP FUNCTION
+  udf_shift_one_day(x INT64) AS (IFNULL((x << 1) & udf_bitmask_lowest_28(),
+      0));
+  --
+CREATE TEMP FUNCTION
+  udf_combine_days(prev INT64,
+    curr INT64) AS (udf_shift_one_day(prev) + IFNULL(curr,
+      0));
+  --
+WITH
+  _current AS (
+  SELECT
+    -- In this raw table, we capture the history of activity over the past
+    -- 28 days for each usage criterion as a single 64-bit integer. The
+    -- rightmost bit represents whether the user was active in the current day.
+    CAST(TRUE AS INT64) AS days_seen_bits,
+    -- We only trust profile_date if it is within one week of the ping submission,
+    -- so we ignore any value more than seven days old.
+    IF(DATE_DIFF(submission_date, profile_date, DAY) BETWEEN 0 AND 6,
+      1 << DATE_DIFF(submission_date, profile_date, DAY),
+      0) AS days_created_profile_bits,
```

Comment:
this can cause an integer overflow if `DATE_DIFF` returns a value too large, and calls `DATE_DIFF` twice, so I recommend this instead:

```
-    IF(DATE_DIFF(submission_date, profile_date, DAY) BETWEEN 0 AND 6,
-      1 << DATE_DIFF(submission_date, profile_date, DAY),
-      0) AS days_created_profile_bits,
+    (1 << MIN(DATE_DIFF(submission_date, profile_date, DAY),
+      7)) & udf_bitmask_lowest_7() AS days_created_profile_bits,
```